change

keleshteri opened the pull request:
change

Commit:
Merge pull request #1 from bumbummen99/update_2

Removed EOL php versions
Added weight option from Crinsane#502
Improved Code quality and testing

bumbummen99 committed:
Adjusted travis.yml

bumbummen99 committed:
removed --prefer-lowest build for testing pruposes

bumbummen99 committed:
Adjust travis.yml

bumbummen99 committed:
Fix travis yml

bumbummen99 committed:
Reverted change to dev dependencies

bumbummen99 committed:
Re add prefer-lowest build

bumbummen99 committed:
revert last change

bumbummen99 committed:
further adjustment

bumbummen99 committed:
Revert "further adjustment"

This reverts commit 22b8d8c6149beeafd7e8485f1e9a683977972797.

bumbummen99 committed:
Revert "revert last change"

This reverts commit 8a73831c33b389a3b9e013c1df6c33438464f7cf.

bumbummen99 committed:
Revert "Re add prefer-lowest build"

This reverts commit a97e055ed3fb4a6d7e9665d76a6bf5659f35b09c.

bumbummen99 committed:
Revert "Reverted change to dev dependencies"

This reverts commit 2e7d9d57305d7418f4063e6be329434bac3aac83.

bumbummen99 committed:
Revert "Fix travis yml"

This reverts commit 5dc86f3c1f75ba3bd8b851188786d6e7042f1950.

bumbummen99 committed:
Revert "Adjust travis.yml"

This reverts commit f80bcc0b6f06e86fce6c1759505fc4a9dae68add.

bumbummen99 committed:
Revert "removed --prefer-lowest build for testing pruposes"

This reverts commit ed298c59d047653e56640a66a620f76d3ec35e49.

bumbummen99 committed:
Revert "Adjusted travis.yml"

This reverts commit cf7cdd01c6c427022494f5e8ff9dd3b63d7ddd0b.

bumbummen99 committed:
Revert "Code formatting & quality"

This reverts commit 52a56768ba81ab47801f2d80be4fe484d27e40ee.

bumbummen99 committed:
Revert "Removed unnecessary assignments"

This reverts commit dbfe5e01fd4c23393fa812388fc9eadef785a55b.

bumbummen99 committed:
Revert "Manually merged https://github.com/Crinsane/LaravelShoppingcart/pull/502"

This reverts commit 5d117a0acd3252faf70a8e11a6707c5df0841eed.

bumbummen99 committed:
Revert "Corrected tavis.yml"

This reverts commit fe2d61160af6bcc74baea43b6a11d78a0f440515.

bumbummen99 committed:
Revert "Removed EOL php versions"

This reverts commit 79b6a5144138596e6d7ee7109d87fee631eeef1e.

bumbummen99 committed:
manuall readded reverted changes

bumbummen99 committed:
Removed Cart facade from coverage (does not provide a public inteterface)
Exclude migrations from coverage
Added tests for InstanceIdentifier
Added InstanceIdentifier support to store, restore and merge
Added weight to CanBeBought trait
Excluded CanBeBought trait from coverage temporarily

bumbummen99 committed:
Added test for model using trait
Added readme for CanBeBoughtTrait
https://github.com/Crinsane/LaravelShoppingcart/issues/427

bumbummen99 committed:
Added styleci.yml

bumbummen99 committed:
Adjustements to code style

bumbummen99 committed:
Further codestyle changes

bumbummen99 committed:
Further adjusting codestyle

bumbummen99 committed:
Further codestyle changes

bumbummen99 committed:
Codestyle

bumbummen99 committed:
Codestyle
ignore tests added to styleci.yml

bumbummen99 committed:
Codestyle

bumbummen99 committed:
Added styleci to readme

bumbummen99 committed:
Added test to ensure everything is ok
https://github.com/Crinsane/LaravelShoppingcart/issues/544

bumbummen99 committed:
Improved Readme
Manually merged https://github.com/Crinsane/LaravelShoppingcart/pull/311

bumbummen99 committed:
removed monitor only master

bumbummen99 committed:
Codestyle

bumbummen99 committed:
Added fields for created and updated at
Added getters
needs testing

bumbummen99 committed:
style

bumbummen99 committed:
Replaced deprecated OR operator https://getcomposer.org/doc/articles/versions.md#version-range
Added 5.8.* compatibility
Updated dependency versions
Added phpunit result cache to gitignore

bumbummen99 committed:
Codestyle
Update version in readme

bumbummen99 committed:
Removed unneccessary return statements

bumbummen99 committed:
Merge pull request #4 from bumbummen99/5.8

Laravel 5.8

bumbummen99 committed:
Remove incompatible versions composer.json

bumbummen99 committed:
Merge branch 'master' into fix_cart_database_timestamps

PazkaL committed:
Update Cart.php, fix CartItem::fromAttributes call

The CartItem::fromAttributes call is missing the $weight parameter, and the CartItemOptions don't work any more because of this.

bumbummen99 committed:
Fixed weight integration, added testcase, fixed weird readme code example

bumbummen99 committed:
Codestyle

bumbummen99 committed:
Merge pull request #5 from PazkaL/patch-1

Update Cart.php, fix CartItem::fromAttributes call

bumbummen99 commented:
I was going to do that on my vacation when i have some more time but it seems like there is a need.

@keleshteri
Can you please change the source branch to backport? This will be the branch to resolve any conflicts.

bumbummen99 committed:
Add validation and exception for weight to CartItem Constructor

bumbummen99 committed:
Added homepages to authors

bumbummen99 committed:
Update CartItem.php

Remove strlen from weight

bumbummen99 committed:
Update CartItem.php

Revert last change

bumbummen99 committed:
Update Cart.php

Changed default value of $weight to 0

bumbummen99 committed:
Update CartTest.php

Added Test for CartItem.php invalid weight exception.

bumbummen99 committed:
Update CartTest.php

Codestyle

bumbummen99 committed:
Update README.md

https://github.com/bumbummen99/LaravelShoppingcart/issues/11
Fixed copypasta in the Readme

bumbummen99 committed:
Update ShoppingcartServiceProvider.php

https://github.com/bumbummen99/LaravelShoppingcart/issues/8
Added missing publish group parameter

bumbummen99 commented:
I have now created a new PR for the correct branch: #571

CaddyDz committed:
fixed typo

bumbummen99 committed:
Merge pull request #15 from CaddyDz/master

fixed typo

dtwebuk committed:
Updated composer.json to allow Laravel 6.0 dependencies

dtwebuk committed:
Replaced array_get with Arr::get as the helper method has been removed in 6.0

bumbummen99 committed:
Updated Laravel version in the readme

bumbummen99 committed:
Merge pull request #21 from dtwebuk/6.0

Updated to support Laravel 6.0

Commit:
Changed $taxRate from private to public, Solve https://github.com/Crinsane/LaravelShoppingcart/issues/436

bumbummen99 committed:
Update composer.json

Should also support Laravel 6.1

Jam-Iko committed:
Add README_uk-UA.md

Jam-Iko committed:
Add Ukrainian translation of README.md - needs proofreading

Jam-Iko committed:
Update README_uk-UA.md

Sartoric committed:
Update CartItem.php

weight variable in if block

Commit:
Corrected PHPDoc

bumbummen99 committed:
Merge pull request #30 from Sartoric/master

Update CartItem.php

bumbummen99 committed:
Merge pull request #28 from Jam-Iko/Readme_uk-UA

README Translation: Ukrainian

tiotobing committed:
Update README_Ind.md

Add Description in Indonesian language, Code Ind/ID

tiotobing committed:
Create README_Idn.md

add descriptions in Indonesian languange, Code Idn/ID

tiotobing committed:
Update README.md

Sartoric committed:
Added a method to erase stored cart

cwprogger committed:
Fix products sequence after changing cart item options

cwprogger committed:
Fix code style

cwprogger committed:
Add a test case to assert items sequence is kept if the options changed

bumbummen99 committed:
Merge pull request #36 from cwprogger/fix-products-sequence

Fix products sequence after changing cart item options

bumbummen99 committed:
Merge pull request #31 from tiotobing/master

Add Description in Indonesian language, Code Ind/ID

bumbummen99 committed:
StyleCI

bumbummen99 committed:
Update Cart.php

Commit:
Dispatch "cart.merged" event,added a parameter to avoid dispatching "cart.added" events while merging

bumbummen99 committed:
Update Cart.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Merge pull request #43 from Sartoric/feature-merged-event

cart merged event

bumbummen99 committed:
Fix link

Commit:
Merge branch 'master' of https://github.com/bumbummen99/LaravelShoppingcart

 Conflicts:
	tests/CartTest.php

Commit:
fix issue with the priceTax value that is not dynamically retrieved

bumbummen99 committed:
Merge pull request #35 from Sartoric/master

Added a method to erase stored cart

Commit:
test added

bumbummen99 committed:
Merge branch 'master' into fix-cartitem-pricetax

bumbummen99 committed:
Merge pull request #56 from Sartoric/fix-cartitem-pricetax

Fix cartitem priceTax bug

bumbummen99 committed:
Correct original auther name

bumbummen99 committed:
Correct original authors name

bumbummen99 committed:
Correct original authors name

Commit:
changed the strategy for cart values calculation (taxes per row),
test,
some fixes

Commit:
added CartItem::priceTotal for rounded total (from internal calculation)

Sartoric committed:
Merge branch 'master' into fix-cart-total-rounding-issue

nexxai committed:
Typofix

bumbummen99 committed:
Merge pull request #60 from nexxai/patch-1

Typofix

Commit:
StyleCI

bumbummen99 committed:
Merge branch 'master' into fix-cart-total-rounding-issue

Commit:
readme

bumbummen99 committed:
Fix # anchor

bumbummen99 committed:
Merge pull request #61 from bumbummen99/readme-cart-total-rounding-issue

Update to readme file

bumbummen99 committed:
Merge branch 'master' into fix-cart-total-rounding-issue

bumbummen99 committed:
Merge pull request #59 from Sartoric/fix-cart-total-rounding-issue

Fix cart total rounding issue

Commit:
Enable gross price as base price

Commit:
SyleCI

royduin committed:
Laravel 7.x support

bumbummen99 committed:
Merge pull request #66 from bumbummen99/feature-gross-price

Enable gross price as base price

bumbummen99 committed:
Merge branch 'master' into patch-1

bumbummen99 committed:
Merge pull request #70 from royduin/patch-1

Laravel 7.x support

bumbummen99 committed:
Update README.md

bumbummen99 committed:
Update README.md

mightyteja committed:
options of an item inside the cart, 

Line 114 to 120

Commit:
Merge branch 'master' into fix_cart_database_timestamps

Commit:
Use Carbon instead of date

Commit:
Make getters public,
Add Test

Commit:
StyleCI

bumbummen99 committed:
Merge pull request #83 from mightyteja/patch-1

options of an item inside the cart,

bumbummen99 committed:
Merge branch 'master' into fix_cart_database_timestamps

Commit:
Improve test,
Fix updatedAt(),

Commit:
Merge branch 'fix_cart_database_timestamps' of github.com:bumbummen99/LaravelShoppingcart into fix_cart_database_timestamps

bumbummen99 committed:
Merge pull request #3 from bumbummen99/fix_cart_database_timestamps

Fix cart database timestamps

bumbummen99 committed:
Update README.md

Minor adjustment to Readme

Commit:
StyleCI

bumbummen99 committed:
Update Cart.php

bumbummen99 committed:
StyleCI

bumbummen99 committed:
Prevent negative subtotal

Manual merge of https://github.com/iainheng/LaravelShoppingcart/commit/33fc42dae115089221f01c2a6a27506b14f763c4

Commit:
Refactor gross price / default  into calculators,
Remove gross_price from config replaced with calculator

Commit:
StyleCI

Commit:
Remove leftover typehint

Commit:
Add priceTotal to CartItemTest

Commit:
Add Calculator Contract,
Add some checking for calculators,

Commit:
Style CI

andcl committed:
Updated README.md documentation

Added getBuyableWeight() method to fully implement the interface, in the Buyable example.

bumbummen99 committed:
Merge pull request #98 from andcl/master

Updated README.md documentation

Commit:
Update for Laravel 8,

Commit:
Added Calculators to the Readme

manojo123 committed:
Update README.md

Added new events description.

manojo123 committed:
Added new events adding, updating, removing.

Commit:
StyleCI

bumbummen99 committed:
Merge pull request #104 from manojo123/master

Add new events (adding, updating, removing)

Commit:
Add stale bot action,

Commit:
Add manual trigger

Commit:
Exclude various labels from stalebot,

olegbespalov committed:
initial config

olegbespalov committed:
change to the ubuntu-18.04

olegbespalov committed:
install packages

olegbespalov committed:
proper caching composer

olegbespalov committed:
fix the spaces

olegbespalov committed:
code coverage action

olegbespalov committed:
pass php version also

olegbespalov committed:
fix the env place

olegbespalov committed:
CI status with the github actions

bumbummen99 committed:
Merge pull request #108 from olegbespalov/master

Migrating to the GitHub Actions

Norris1z committed:
Update Cart.php

Add instance to storedCartWithIdentifierExists

Norris1z committed:
Use instance in query if there are more than one instances and add tests

Norris1z committed:
Style CI fixes and remove assertDatabaseCount

Norris1z committed:
Style CI fixes

Norris1z committed:
Style CI fixes

Norris1z committed:
rename the storedCartWithIdentifier method to storedCartInstanceWithIdentifierExists, changed the signature of the merge and storedCartInstanceWithIdentifierExists methods and add instance to all stored cart retrievals

Norris1z committed:
Add savedcartinstance as a parameter to the merge documentation

Norris1z committed:
Style CI Fixes

Norris1z committed:
Style CI Fixes

Norris1z committed:
Add more tests

Norris1z committed:
Style CI fixes

ganyicz committed:
Add macroable trait

Commit:
Merge pull request #113 from thefilip/macroable

Add macroable trait

Norris1z committed:
Merge branch 'master' into master

Commit:
Merge pull request #111 from Norris1z/master

Use current instance when checking if storedCartWithIdentifierExists

Commit:
Rename countInstances and improve PHPDoc,

Commit:
Update CartTest.php

geisi committed:
Fixing InvalidArgumentException when adding buyable models to Cart on newer Laravel Versions.

bumbummen99 committed:
Update CanBeBought.php

Use getAttribute method from HasAttributes trait,

bumbummen99 committed:
Update CanBeBought.php

Add missing assignment operator

bumbummen99 committed:
Update CanBeBought.php

Add missing bracket

bumbummen99 committed:
Update BuyableProduct.php

Extend Model

bumbummen99 committed:
Update BuyableProductTrait.php

Extend default Model

bumbummen99 committed:
Update BuyableProduct.php

bumbummen99 committed:
Update BuyableProductTrait.php

bumbummen99 committed:
Update BuyableProductTrait.php

bumbummen99 committed:
Update BuyableProduct.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update BuyableProduct.php

bumbummen99 committed:
Update BuyableProductTrait.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update BuyableProduct.php

bumbummen99 committed:
Update BuyableProductTrait.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update BuyableProductTrait.php

bumbummen99 committed:
Update BuyableProduct.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update BuyableProduct.php

bumbummen99 committed:
Update BuyableProduct.php

bumbummen99 committed:
Update BuyableProductTrait.php

bumbummen99 committed:
Update BuyableProductTrait.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update BuyableProductTrait.php

Commit:
StyleCI

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
Update CartTest.php

bumbummen99 committed:
StyleCI

Commit:
Merge pull request #115 from geisi/master

Fixing InvalidArgumentException when adding buyable models

rolandassale committed:
Fix ttle variable error

bumbummen99 committed:
Merge pull request #118 from guysolamour/patch-1

Fix ttle variable error

bumbummen99 committed:
Update README.md

bumbummen99 committed:
Create contributors.yml

bumbummen99 committed:
Update README.md

Commit:
contrib-readme-action has updated readme

Commit:
Make the CartItem aware of its cart instance

Commit:
Use the helper method

Commit:
Fixing continous-integration complaint

macbookandrew committed:
fix docs

macbookandrew committed:
add handling for arrays

bumbummen99 committed:
Merge pull request #141 from macbookandrew/bugfix/cart-item-to-array

fix CartItem options

Commit:
contrib-readme-action has updated readme

bumbummen99 committed:
Update CartItem.php

bumbummen99 committed:
Merge branch 'master' into master

bumbummen99 committed:
Merge pull request #140 from ChrisThompsonTLDR/master

Add a cart instance string property on CartItems

bumbummen99 committed:
Update README.md

bumbummen99 committed:
Update contributors.yml

github-actions[bot] committed:
contrib-readme-action has updated readme

bumbummen99 committed:
Merge pull request #143 from bumbummen99/contributors-readme-action-M-4_Tz-nNv

contributors readme action update

github-actions[bot] committed:
contrib-readme-action has updated readme

bumbummen99 committed:
Merge pull request #144 from bumbummen99/contributors-readme-action-MK_QGGzZOr

contributors readme action update

bumbummen99 committed:
Update composer.json

bumbummen99 committed:
Update composer.json

quintenbuis committed:
Add Laravel 9 compatibility

bumbummen99 committed:
Merge pull request #165 from quintenbuis/master

Add Laravel 9 compatibility

github-actions[bot] committed:
contrib-readme-action has updated readme

bumbummen99 committed:
Merge pull request #166 from bumbummen99/contributors-readme-action-meFlvdHOi1

contributors readme action update

mauriciv committed:
Use base64 to avoid truncation with PostgreSQL

bumbummen99 committed:
Merge pull request #167 from mauriciv/master

Use base64 to avoid truncation with PostgreSQL

github-actions[bot] committed:
contrib-readme-action has updated readme

bumbummen99 committed:
Merge pull request #168 from bumbummen99/contributors-readme-action-KhPaPlWC0I

contributors readme action update

bumbummen99 committed:
Update README.md

bumbummen99 committed:
Update README.md

sharif01946 commented:
upgrade verssion
